api/src/test/java/org/pmiops/workbench/api/UserControllerTest.java

```diff
@@ -252,93 +262,214 @@ public void testUserSort() {
     assertThat(robinsonsAsc.getUsers()).containsAllIn(newAscending).inOrder();
   }
 
+  // Combinatorial tests for listBillingAccounts:
```

Comment:
Yeah, yeah, I know.  However:
* the logic is obtuse enough that we should be explicit here
* half of these will go away after enableBillingUpgrade is permanently set to true

If I follow up on the BillingAccountService idea, these would move to a new BillingAccountServiceTest

Reply:
Note: This is an ideal scenario for a `@Parameterized` test case. We don't have the pattern established in the codebase though and the first usage will probably be a substantial undertaking, so not for this PR.